### PR TITLE
Proper input value type verification for `append` method

### PR DESCRIPTION
**Pull request by Raalsky**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **asked the docs owner** to review all the updated user-facing interfaces?

### CODE REVIEW

File: src/neptune/handler.py [hunk 1]
```diff
@@ -681,12 +681,7 @@ def validate_and_transform_to_extend_format(value):
         so work can be delegated to `extend` method."""
```
Comment (shnela) [inline src/neptune/handler.py:681]: Can we change `validate_and_transform_to_extend_format` to `transform_to_extend_format`?

File: tests/unit/neptune/new/test_handler.py [hunk 2]
```diff
@@ -270,15 +271,15 @@ def test_log_many_values(self):
 
     def test_append_many_values_cause_error(self):
```
Comment (shnela) [inline tests/unit/neptune/new/test_handler.py:272]: It sould not be called `test_append_many_values_cause_error`, now error is caused by non-supported value.
We could test some `set`, `tuple` and instance of custom class here as well.

File: src/neptune/handler.py (hunk 1 above)
Comment (Raalsky) [reply src/neptune/handler.py:681]: Done

File: tests/unit/neptune/new/test_handler.py (hunk 2 above)
Comment (Raalsky) [reply tests/unit/neptune/new/test_handler.py:276]: Done

**Review by shnela — APPROVED** [review]

**Comment by Raalsky** [comment]
@normandy7, we had a bug that whenever the user was calling the `.append(...)` method with list/tuple/list we were raising an exception about provided collection (probably to differentiate it from `extend`). In the spec, the `append` method should accept a dictionary (of namespaces) or any other value as a single. There was no need for such a check.

**Comment by normandy7** [comment]
> @normandy7, we had a bug that whenever the user was calling the `.append(...)` method with list/tuple/list we were raising an exception about provided collection (probably to differentiate it from `extend`). In the spec, the `append` method should accept a dictionary (of namespaces) or any other value as a single. There was no need for such a check. Probably no action is needed on your side.

Got it. So this was not possible before:

```
run["train"].append({"acc": acc, "loss": loss})
```

but should be OK after this fix?

**Comment by Raalsky** [comment]
Not exactly.:
`run["train"].append({"acc": acc, "loss": loss})`:
- Before: Works as expected
- After this PR: Still works


`run['a'].append((1, 2))` / `run['a'].append([3, 4])` / `run['a'].append({5, 6})`:
- Before: raises an exception about being collection and invalid type provided
- After this PR: If we'll publish it somehow before 1.0.0 then it would be just translated to string without any warning. After 1.0.0 this will warn with the default `NeptuneUnsupportedType` warning.

**Comment by normandy7** [comment]
> * After this PR: If we'll publish it somehow before 1.0.0 then it would be just translated to string without any warning. After 1.0.0 this will warn with the default `NeptuneUnsupportedType` warning.

Ah, makes sense.

**Pull request merged**